wire-ios-sync-engine/Tests/Source/Use cases/SetAllowGuestsAndServicesUseCaseTests.swift

```diff
@@ -0,0 +1,180 @@
+            }
+
+            wait(for: [expectation], timeout: 4.0)
+        }
+    }
+
+    func testGuestEnablementFails_WithInsufficientPermissions() async {
+        await syncContext.perform { [self] in
+            // GIVEN
+            let role = Role.insertNewObject(in: syncContext)
+            let action = Action.insertNewObject(in: syncContext)
+
+            action.name = "modify_conversation_access"
+            role.actions = [action]
+            mockConversation.addParticipantAndUpdateConversationState(user: mockSelfUser, role: role)
+
+            let mockHandler = MockActionHandler<SetAllowGuestsAndServicesAction>(result: .failure(.unknown), context: syncContext.notificationContext)
+
+            let expectation = XCTestExpectation(description: "Completion should be called with a failure due to insufficient permissions")
+
+            // WHEN
+            sut.invoke(conversation: mockConversation, allowGuests: true, allowServices: false) { result in
+                // THEN
+                switch result {
+                case .success:
+                    XCTFail("Expected operation to fail, but it succeeded.")
+                case .failure(let error):
+                    print("Operation failed")
+                }
+                expectation.fulfill()
+            }
+
+            wait(for: [expectation], timeout: 4.0)
+        }
+    }
+
+    func testServicesEnablementSucceeds() async {
+
+        await syncContext.perform { [self] in
+            // GIVEN
+            let role = Role.insertNewObject(in: syncContext)
+            let action = Action.insertNewObject(in: syncContext)
+            action.name = "modify_conversation_access"
+            role.actions = [action]
+
+            mockConversation.addParticipantAndUpdateConversationState(user: mockSelfUser, role: role)
+
+            let mockHandler = MockActionHandler<SetAllowGuestsAndServicesAction>(result: .success(()), context: syncContext.notificationContext)
+
+            let expectation = XCTestExpectation(description: "completion should be called")
+
+            // WHEN
+            sut.invoke(conversation: mockConversation, allowGuests: false, allowServices: true) { result in
+                // THEN
+                switch result {
+                case .success:
+                    print("Operation successful")
+                case .failure(let error):
+                    XCTFail("Test failed with error: \(error)")
+                }
+
+                expectation.fulfill()
+            }
+
+            wait(for: [expectation], timeout: 4.0)
+        }
+    }
+
+    func testServicesEnablementFails_WithInsufficientPermissions() async {
+
+        await syncContext.perform { [self] in
+            // GIVEN
+            let role = Role.insertNewObject(in: syncContext)
+            let action = Action.insertNewObject(in: syncContext)
+
+            action.name = "modify_conversation_access"
+            role.actions = [action]
```

Comment:
This looks the same as the test where the user has permission...

Reply:
Clean up the entire test class. I think now it's much more clear: https://github.com/wireapp/wire-ios/commit/80a04a5d4aaf473f97eee2a7d49763ac74f136c2